code/cycler/src/wattrex_battery_cycler/mid/mid_str/mid_str_node.py

```diff
@@ -0,0 +1,166 @@
+            exp_info, battery_info, profile_info = self.db_iface.get_start_queued_exp()
+            self.__actual_exp_id = exp_info.exp_id
+            log.debug("Sending new experiment to APP_MANAGER")
+            self.str_data.send_data(MidStrCmdDataC(cmd_type= MidStrDataCmdE.EXP_DATA,
+                    experiment= exp_info, battery= battery_info, profile= profile_info))
+        elif command.cmd_type == MidStrReqCmdE.GET_EXP_STATUS:
+            exp_status = self.db_iface.get_exp_status(exp_id= self.__actual_exp_id)
+            self.str_data.send_data(MidStrCmdDataC(cmd_type= MidStrDataCmdE.EXP_STATUS,
+                                                   exp_status= exp_status))
+        elif command.cmd_type == MidStrReqCmdE.GET_CS:
+            cycler_info = self.db_iface.get_cycler_station_info()
+            self.str_data.send_data(MidStrCmdDataC(cmd_type= MidStrDataCmdE.CS_DATA,
+                                                   station= cycler_info))
+        elif command.cmd_type == MidStrReqCmdE.SET_EXP_STATUS and command.exp_status is not None:
+            self.db_iface.modify_current_exp(exp_status= command.exp_status,
+                                             exp_id= self.__actual_exp_id)
+        else:
+            log.error(("Can`t apply command. Error in command format, "
+                       "check command type and payload type"))
+
+    def sync_shd_data(self) -> None:
+        '''Update local data
+        '''
+        self.db_iface.all_status: CyclerDataAllStatusC = self.globlal_all_status.read()
+        self.db_iface.gen_meas: CyclerDataGenMeasC     = self.globlal_gen_meas.read()
+        self.db_iface.ext_meas: CyclerDataExtMeasC     = self.globlal_ext_meas.read()
+
+    def stop(self) -> None:
+        """Stop the node if it is not already closed .
+        """
+        self.db_iface.close_db_connection()
+        self.working_flag.clear()
+        self.status = SysShdNodeStatusE.STOP
+        log.critical(f"Stopping {current_thread().name} node")
+
+    def process_iteration(self) -> None:
+        """AI is creating summary for process_iteration
+        """
+        try:
+            # Syncronising shared data
+            self.sync_shd_data()
+            # Receive and write alarms
+            self.__receive_alarms()
+
+            if len(self.__new_raised_alarms)>0:
+                self.db_iface.write_new_alarm(alarms= self.__new_raised_alarms,
+                                              exp_id= self.__actual_exp_id)
+                self.__new_raised_alarms.clear()
+            # log.debug("+++++ After write alarams in db_iface object +++++")
+            ### Write measures and status changes
+            if self.db_iface.gen_meas.instr_id is not None:
+                self.db_iface.write_generic_measures(exp_id= self.__actual_exp_id)
+                log.critical("TEST COMMITING CHANGES")
+                self.db_iface.commit_changes()
```

Comment:
Add timeout function

Reply:
Done

Reply:
This commit is for test, already removed